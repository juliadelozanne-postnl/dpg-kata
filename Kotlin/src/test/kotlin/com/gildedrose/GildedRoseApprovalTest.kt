package com.gildedrose

import org.approvaltests.combinations.CombinationApprovals
import org.junit.jupiter.api.Test

internal class GildedRoseApprovalTest {

    @Test
    fun `combination approval tests for aged brie items`() {
        fun updateQualityItem(sellIn: Int, quality: Int): String {
            val items = listOf(AgedBrie(sellIn, quality))
            val app = GildedRose(items)
            app.updateItems()
            return items[0].toString()
        }

        val sellIns = arrayOf(0, 6, 11, 12)
        val qualities = arrayOf(0, 10, 49, 50)

        CombinationApprovals.verifyAllCombinations(::updateQualityItem, sellIns, qualities)
    }

    @Test
    fun `combination approval tests for conjured items`() {
        fun updateQualityItem(sellIn: Int, quality: Int): String {
            val items = listOf(ConjuredItem(sellIn, quality))
            val app = GildedRose(items)
            app.updateItems()
            return items[0].toString()
        }

        val sellIns = arrayOf(0, 6, 11, 12)
        val qualities = arrayOf(0, 10, 49, 50)

        CombinationApprovals.verifyAllCombinations(::updateQualityItem, sellIns, qualities)
    }

    /*@Test
    fun `update quality approval test`() {
        fun updateQualityItem(name: String, sellIn: Int, quality: Int): String {
            val items = listOf(Item(name, sellIn, quality))
            val app = GildedRose(items)
            app.updateQuality()
            return items[0].toString()
        }

        val sellIns = arrayOf(0, 6, 11, 12)
        val qualities = arrayOf(0, 10, 49, 50)
        val names = arrayOf(
            "Sulfuras, Hand of Ragnaros",
            "Aged Brie",
            "Backstage passes to a TAFKAL80ETC concert",
            "+5 Dexterity Vest",
            "Elixir of the Mongoose"
        )

        CombinationApprovals.verifyAllCombinations(::updateQualityItem, names, sellIns, qualities)
    }*/
}
