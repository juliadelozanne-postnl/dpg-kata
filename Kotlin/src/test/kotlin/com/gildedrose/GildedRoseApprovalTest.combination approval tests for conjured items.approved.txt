[0, 0] => Conjured Mana Cake, -1, 0 
[0, 10] => Conjured Mana Cake, -1, 6 
[0, 49] => Conjured Mana Cake, -1, 45 
[0, 50] => Conjured Mana Cake, -1, 46 
[6, 0] => Conjured Mana Cake, 5, 0 
[6, 10] => Conjured Mana Cake, 5, 8 
[6, 49] => Conjured Mana Cake, 5, 47 
[6, 50] => Conjured Mana Cake, 5, 48 
[11, 0] => Conjured Mana Cake, 10, 0 
[11, 10] => Conjured Mana Cake, 10, 8 
[11, 49] => Conjured Mana Cake, 10, 47 
[11, 50] => Conjured Mana Cake, 10, 48 
[12, 0] => Conjured Mana Cake, 11, 0 
[12, 10] => Conjured Mana Cake, 11, 8 
[12, 49] => Conjured Mana Cake, 11, 47 
[12, 50] => Conjured Mana Cake, 11, 48 
